model_analyzer/triton/model/model_config.py

```diff
@@ -258,14 +258,56 @@ def set_cpu_only(self, cpu_only):
 
     def cpu_only(self):
         """
-        Return
+        Returns
         -------
         bool
             Whether the model should be run on CPU only
         """
 
         return self._cpu_only
 
+    def is_ensemble(self) -> bool:
+        """
+        Returns
+        -------
+        bool
+           True if this is an ensemble model
+        """
+
+        return getattr(self._model_config, "platform") == "ensemble"
+
+    def get_ensemble_submodels(self) -> Optional[List[str]]:
+        """
+        Returns
+        -------
+            List[str]: Sub-model names
+        """
+
+        if not self.is_ensemble():
+            return None
+
+        submodels = [
+            model['modelName']
+            for model in self.to_dict()['ensembleScheduling']['step']
+        ]
+        return submodels
+
+    def set_submodel_variant_name(self, submodel_name: str,
+                                  variant_name: str) -> None:
+        """
+        Replaces the Ensembles submodel's name with the variant name
+        """
+
+        if not self.is_ensemble():
+            return
```

Comment:
Can we error out here? that would mean bad code and I don't want to silently do that

Reply:
Changed this to raise an TMA exception.

Reply:
Add a test to protect that and I think this is ready to be merged. (create an ensemble config that is missing 'step', and expect assert to be raised)